test/publish.sh

```diff
@@ -27,55 +27,66 @@ else
   echo "No credential helpers found, push to docker may not function properly"
 fi
 
-DOCKER_HUB=${DOCKER_HUB:-gcr.io/istio-testing}
 GCS_BUCKET=${GCS_BUCKET:-istio-build/test}
 HELM_BUCKET=${HELM_BUCKET:-istio-build/test/charts}
-VERSION="1.14.0-releasebuilder.$(git rev-parse --short HEAD)"
 COSIGN_KEY=${COSIGN_KEY:-}
+GITHUB_ORG=${GITHUB_ORG:-istio}
+DOCKER_HUB=${DOCKER_HUB:-gcr.io/istio-testing}
+
+if [[ -n ${ISTIO_ENVOY_BASE_URL:-} ]]; then
+  PROXY_OVERRIDE="proxyOverride: ${ISTIO_ENVOY_BASE_URL}"
+fi
+
+VERSION="1.14.0-releasebuilder.$(git rev-parse --short HEAD)"
```

Comment:
I suspect this should also match the version associated with the branch in question, 1.17 for the main branch, 1.16 for this branch. One of the values that isn't getting updated, although other values here seem to have been updated. Have to do more research, but still a number of new 1.14.0-release-builders images created like here: https://console.cloud.google.com/gcr/images/istio-testing/global/istioctl

Reply:
- release-1.17 branch does not exist for release-builder yet
- the current version in 1.16 is set to 1.14. It seems to me that this test is out of date and is similar enough to the build.sh script that it should be removed and functionality combined with build.sh. 
https://github.com/istio/release-builder/blob/release-1.16/test/publish.sh#L33
-- updated VERSION to be 1.16

Reply:
The release-1.17 branch is currently master (will be cut in January). The process is for changes to be made in master and then cherry-picked. I suspect we need this change in master as well, although it will be a manual cherry-pick with the version.

Reply:
I just closed this PR and made a new clean on on master. thank you for pointing that out.
https://github.com/istio/release-builder/pull/1319